File: libs/sdk-typescript/src/utils/Path.ts
/*
 * Copyright 2025 Daytona Platforms Inc.
 * SPDX-License-Identifier: Apache-2.0
 */

import * as pathe from 'pathe'

export function prefixRelativePath(prefix: string, path?: string): string {
  let result = prefix

  if (path) {
    path = path.trim()
    if (path === '~') {
      result = prefix
    } else if (path.startsWith('~/')) {
      result = pathe.join(prefix, path.slice(2))
    } else if (pathe.isAbsolute(path)) {
      result = path
    } else {
      result = pathe.join(prefix, path)
    }
  }

  return result
}

File: libs/sdk-python/src/daytona/_utils/path.py
# Copyright 2025 Daytona Platforms Inc.
# SPDX-License-Identifier: Apache-2.0

import os
from pathlib import PurePosixPath
from typing import Optional


def prefix_relative_path(prefix: str, path: Optional[str] = None) -> str:
    result = prefix

    if path:
        path = path.strip()
        if path == "~":
            result = prefix
        elif path.startswith("~/"):
            result = os.path.join(prefix, path[2:])
        elif PurePosixPath(path).is_absolute():
            result = path
        else:
            result = os.path.join(prefix, path)

    return result
